agent/consul/state/catalog.go

```diff
@@ -3057,24 +3075,27 @@ func (s *Store) ServiceTopology(
 	}
 
 	downstreamDecisions := make(map[string]structs.IntentionDecisionSummary)
-	for _, svc := range intentionDownstreams {
-		if _, ok := seenDownstreams[svc.Name.String()]; ok {
-			// Avoid duplicating entry
-			continue
-		}
-		downstreamNames = append(downstreamNames, svc.Name)
-		downstreamDecisions[svc.Name.String()] = svc.Decision
-
-		var source string
-		switch {
-		case svc.Decision.HasExact:
-			source = structs.TopologySourceSpecificIntention
-		case svc.Decision.DefaultAllow:
-			source = structs.TopologySourceDefaultAllow
-		default:
-			source = structs.TopologySourceWildcardIntention
+	// Only transparent proxies have downstreams from intentions
+	if hasTransparent {
```

Comment:
Sorry, I think I miscommunicated here.

A service being a transparent proxy only influences its upstreams, not its downstreams.

This means that a downstream is a downstream from intentions if the downstream is a transparent proxy. hasTransparent on the other hand is based on the target service.

I think what would clarify this is to look at the usage and comment for `tproxyMap` below:

> // Store downstreams with at least one instance in transparent proxy mode.
> // This is to avoid returning downstreams from intentions when none of the downstreams are transparent proxies.